src/publisher/Publisher.ts

```diff
@@ -178,12 +161,28 @@ export class Publisher extends OTEventEmitter<{
     }
   }
 
-  destroy(): void {
+  destroy(): this {
```

Comment:
This had the wrong return type according to [the docs](https://tokbox.com/developer/sdks/js/reference/Publisher.html#destroy:~:text=promise%20is%20rejected.-,destroy,-()%20%E2%86%92%20%7BPublisher). I must have accidentally mixed up these docs with `Session.destroy()`.

Upon digging deeper, the user doesn't _leave_ the call when the publisher is destroyed. The publisher stops publishing locally, and the DOM elements are removed.

This manifests in the network test because `Session.destroy()` calls `call.leave()` in the shim. However, if a user called `publisher.destroy()` in the previous code, they'll have _left_ the Daily call before `session.destroy()` was called, causing the callback to never get fired in `session.destroy()`